src/asm/fstack.c

```diff
@@ -240,7 +240,9 @@ bool yywrap(void)
 
 		// If this is a FOR, update the symbol value
 		if (contextStack->forName && fileInfo->iters[0] <= contextStack->nbReptIters) {
-			contextStack->forValue += contextStack->forStep;
+			// Avoid arithmetic overflow runtime error
```

Comment:
rpn.c already does addition of user-defined variables as unsigned, for exactly this reason. We just had missed that case in `FOR` loops so far.

Reply:
Thanks, I've added an explicit `int32_t` cast.

Reply:
Adding a cast doesn't fix the situation: it's the conversion that is UB, not the fact that it is implicit. (C doesn't make a difference between implicit and explicit conversions.)